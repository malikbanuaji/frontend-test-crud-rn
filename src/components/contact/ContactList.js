import React from 'react';
import {FlatList, View} from 'react-native';
import {Spacing} from '../../styles';
import ContactListItem from '../common/ContactListItem';

function ContactList({refreshing, onRefresh, data, onPressItem}) {
  function _renderItem({item}) {
    return (
      <ContactListItem
        id={item.id}
        firstName={item.firstName}
        lastName={item.lastName}
        age={item.age}
        photo={item.photo}
        onPress={onPressItem}
      />
    );
  }

  function _keyExtractor(item) {
    return item.id;
  }

  return (
    <FlatList
      testID={'contact-list'}
      refreshing={refreshing}
      onRefresh={onRefresh}
      data={data}
      renderItem={_renderItem}
      keyExtractor={_keyExtractor}
      ListFooterComponent={<View style={{marginBottom: Spacing.largest}} />}
    />
  );
}

export default ContactList;
