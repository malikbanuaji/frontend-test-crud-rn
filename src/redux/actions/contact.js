export const SET_CONTACT_LIST = 'SET_CONTACT_LIST';
export const ADD_NEW_CONTACT = 'ADD_NEW_CONTACT';
export const UPDATE_CONTACT = 'UPDATE_CONTACT';
export const DELETE_CONTACT = 'DELETE_CONTACT';

/**
 * Set contact list
 * @param {Object} payload
 * @param {any[]} payload.data List of contact
 */
export const setContactList = (payload) => ({
  type: SET_CONTACT_LIST,
  payload,
});

/**
 * Add new contact
 * @param {Object} payload
 * @param {string} payload.id - id of new contact
 * @param {string} payload.firstName - firstName of new contact
 * @param {string} payload.lastName - lastName of new contact
 * @param {number} payload.age - age of new contact
 * @param {string} payload.photo - url of photo of new contact or N/A if none
 */
export const addNewContact = (payload) => ({
  type: ADD_NEW_CONTACT,
  payload,
});

/**
 * Update contact
 * @param {Object} payload
 * @param {string} payload.id - id of existing contact
 * @param {string} payload.firstName - firstName of existing contact
 * @param {string} payload.lastName - lastName of existing contact
 * @param {number} payload.age - age of existing contact
 * @param {string} payload.photo - url of photo of existing contact or N/A if none
 */
export const updateContact = (payload) => ({
  type: UPDATE_CONTACT,
  payload,
});

/**
 * Delete contact
 * @param {Object} payload
 * @param {string} payload.id Id's of contact that want to be deleted
 */
export const deleteContact = (payload) => ({
  type: DELETE_CONTACT,
  payload,
});
