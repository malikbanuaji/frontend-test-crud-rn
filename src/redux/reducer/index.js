import ContactReducer from './contact';

export {ContactReducer};
