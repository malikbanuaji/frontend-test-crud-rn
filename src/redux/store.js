const {createStore, combineReducers} = require('redux');
const {ContactReducer} = require('./reducer');

function configureStore() {
  const rootReducer = combineReducers({
    contact: ContactReducer,
  });
  const store = createStore(rootReducer);
  return store;
}

export {configureStore};
