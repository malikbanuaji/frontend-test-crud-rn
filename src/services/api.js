import * as Settings from './settings';
import Axios from 'axios';

const config = {
  baseURL: Settings.BASE_URL,
  timeout: Settings.TIMEOUT,
};

export async function fetchAllContact() {
  return await Axios.get('/contact', config);
}

export async function fetchContact(id) {
  return await Axios.get(`/contact/${id}`, config);
}

export async function addContact({firstName, lastName, age, photo}) {
  return await Axios.post(
    '/contact',
    {firstName, lastName, age, photo},
    config,
  );
}

export async function editContact({id, firstName, lastName, age, photo}) {
  return await Axios.put(
    `/contact/${id}`,
    {firstName, lastName, age, photo},
    config,
  );
}

export async function deleteContact(id) {
  return await Axios.delete(`/contact/${id}`, config);
}
