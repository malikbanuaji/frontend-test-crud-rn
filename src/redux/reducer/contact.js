import * as Actions from '../actions/contact';

const initialState = {
  data: [],
};

export default (state = initialState, {type, payload}) => {
  switch (type) {
    case Actions.SET_CONTACT_LIST:
      return {...state, data: payload.data};
    case Actions.DELETE_CONTACT:
      return {
        ...state,
        data: state.data.filter((value) => {
          return value.id !== payload.id;
        }),
      };
    case Actions.ADD_NEW_CONTACT:
      return {
        ...state,
        data: [...state.data, payload],
      };
    case Actions.UPDATE_CONTACT:
      return {
        ...state,
        data: state.data.map((value) => {
          if (value.id === payload.id) {
            return payload;
          }
          return value;
        }),
      };

    default:
      return state;
  }
};
