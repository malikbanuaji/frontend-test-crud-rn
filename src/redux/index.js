import {configureStore} from './store';

export {configureStore};
