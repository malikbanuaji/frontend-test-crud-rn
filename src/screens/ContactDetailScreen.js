import {useNavigation, useRoute} from '@react-navigation/native';
import React, {useMemo, useState} from 'react';
import {Alert, StyleSheet, Text, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import BackButton from '../components/common/BackButton';
import DangerButton from '../components/common/DangerButton';
import HeaderBar from '../components/common/HeaderBar';
import HeaderLoading from '../components/common/HeaderLoading';
import NormalButton from '../components/common/NormalButton';
import PhotoBig from '../components/common/PhotoBig';
import * as ContactActions from '../redux/actions/contact';
import {Api} from '../services';
import {Spacing, Typography} from '../styles';
import Utils from '../utils';
import FullName from '../components/common/FullName';

export default function ContactDetailScreen() {
  const dispatch = useDispatch();
  const navigation = useNavigation();
  const route = useRoute();

  const _id = route.params?.id;
  const contactList = useSelector((state) => state.contact.data);

  const contactDetail = useMemo(() => {
    const _detail = contactList.find((value) => value.id === _id) || {};

    return {
      id: _id,
      photo: _detail.photo,
      firstName: _detail.firstName,
      lastName: _detail.lastName,
      age: _detail.age,
    };
  }, [_id, contactList]);

  function _handleBackButton() {
    navigation.goBack();
  }

  const [isLoading, setIsLoading] = useState(false);

  function deleteContact() {
    setIsLoading(true);
    Api.deleteContact(_id)
      .then(() => {
        setIsLoading(false);
        dispatch(ContactActions.deleteContact({id: _id}));
        navigation.goBack();
      })
      .catch((e) => {
        setIsLoading(false);
        Utils.SmallMessage.showError();
      });
  }

  function _handlePressDelete() {
    Alert.alert('Hapus Kontak', 'Kontak ini akan dihapus', [
      {
        text: 'BATAL',
      },
      {
        text: 'HAPUS',
        onPress: deleteContact,
      },
    ]);
  }

  function _handlePressEdit() {
    navigation.navigate('CreateEditContact', {
      actionStatus: 'EDIT',
      contactDetail: contactDetail,
    });
  }

  return (
    <View>
      <HeaderBar>
        {isLoading ? (
          <HeaderLoading text={'Menghapus kontak...'} />
        ) : (
          <BackButton onPress={_handleBackButton} />
        )}
      </HeaderBar>
      <View style={styles.photoButtonView}>
        <PhotoBig
          photo={contactDetail.photo}
          firstName={contactDetail.firstName}
          lastName={contactDetail.lastName}
        />
      </View>
      <View style={styles.contentContainer}>
        <FullName
          firstName={contactDetail.firstName}
          lastName={contactDetail.lastName}
        />
        <Text key={'age'} style={styles.age}>
          {`${contactDetail.age}`}
        </Text>
      </View>
      <View style={styles.buttonContainer}>
        <NormalButton
          disabled={isLoading === true}
          style={styles.button}
          text={'Ubah kontak'}
          onPress={_handlePressEdit}
        />
        <DangerButton
          disabled={isLoading === true}
          style={styles.button}
          text={'Hapus kontak'}
          onPress={_handlePressDelete}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  contentContainer: {
    marginVertical: Spacing.base,
  },
  photoButtonView: {
    marginBottom: Spacing.base,
    alignItems: 'center',
  },

  age: {
    ...Typography.contactSubtitle,
    marginHorizontal: Spacing.base,
  },
  buttonContainer: {
    marginTop: Spacing.smaller,
  },
  button: {
    marginHorizontal: Spacing.base,
    marginVertical: Spacing.smaller,
  },
});
