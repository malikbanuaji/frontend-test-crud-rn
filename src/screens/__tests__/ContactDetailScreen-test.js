import {cleanup, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import * as ReactNative from 'react-native';
import {Alert, Platform} from 'react-native';
import {Provider} from 'react-redux';
import {act} from 'react-test-renderer';
import {configureStore} from '../../redux';
import {setContactList} from '../../redux/actions/contact';
import {Api} from '../../services';
import ContactDetailScreen from '../ContactDetailScreen';

jest.doMock('react-native', () => {
  // Extend ReactNative
  return Object.setPrototypeOf(
    {
      // Redefine an export, like a component
      Button: 'MockedButton',

      // Mock out properties of an already mocked export
      LayoutAnimation: {
        ...ReactNative.LayoutAnimation,
        configureNext: jest.fn(),
      },

      // Mock a native module
      NativeModules: {
        ...ReactNative.NativeModules,
        Override: {great: 'success'},
      },

      Alert: {
        ...ReactNative.Alert,
        alert: jest.fn(),
      },

      ToastAndroid: {
        ...ReactNative.ToastAndroid,
        show: jest.fn(),
      },

      Platform: {
        ...ReactNative.Platform,
        OS: 'android',
        Version: 23,
        isTesting: true,
        select: (objs) => objs['android'],
      },
    },
    ReactNative,
  );
});

jest.mock('../../services');

jest.mock('react-native-vector-icons/Feather', () => 'Icon');

describe('ContactList', () => {
  let navigation;
  Platform.OS = 'android';

  const store = configureStore();
  // init state
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);

  beforeEach(() => {
    jest.spyOn(Alert, 'alert');
    Api.deleteContact.mockReset();
    Platform.OS = 'android';
    navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };

    store.dispatch(
      setContactList({
        data: [
          {
            id: '93ad6070-c92b-11e8-b02f-cbfa15db428b',
            firstName: 'Frodo',
            lastName: 'Baggins',
            age: 24,
            photo: 'N/A',
          },
          {
            id: 'b3abd640-c92b-11e8-b02f-cbfa15db428b',
            firstName: 'Frodo',
            lastName: 'Baggins',
            age: 24,
            photo:
              'http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550',
          },
        ],
      }),
    );
  });

  afterEach(() => {
    jest.clearAllTimers();
    cleanup();
  });

  it('loads contact with given id on route parameter', async () => {
    Api.deleteContact.mockResolvedValue({
      data: {},
    });
    const route = {
      params: {
        id: '93ad6070-c92b-11e8-b02f-cbfa15db428b',
      },
    };
    const component = (
      <Provider store={store}>
        <ContactDetailScreen navigation={navigation} route={route} />
      </Provider>
    );

    const {toJSON} = render(component);
    expect(toJSON()).toMatchSnapshot();
  });

  it('loads contact with given id on route parameter (with picture)', async () => {
    Api.deleteContact.mockResolvedValue({
      data: {},
    });
    const route = {
      params: {
        id: 'b3abd640-c92b-11e8-b02f-cbfa15db428b',
      },
    };
    const component = (
      <Provider store={store}>
        <ContactDetailScreen navigation={navigation} route={route} />
      </Provider>
    );

    const {toJSON} = render(component);
    expect(toJSON()).toMatchSnapshot();
  });

  it('handle delete contact button correctly', async () => {
    Api.deleteContact.mockResolvedValue({
      data: {},
    });
    const route = {
      params: {
        id: 'b3abd640-c92b-11e8-b02f-cbfa15db428b',
      },
    };
    const component = (
      <Provider store={store}>
        <ContactDetailScreen navigation={navigation} route={route} />
      </Provider>
    );

    const {getByText} = render(component);
    const deleteButton = getByText('Hapus kontak');
    fireEvent.press(deleteButton.parent.parent);
    expect(Alert.alert).toHaveBeenCalledWith(
      expect.any(String),
      expect.any(String),
      expect.objectContaining([
        {
          text: 'BATAL',
        },
        {
          text: 'HAPUS',
          onPress: expect.any(Function),
        },
      ]),
    );
    await act(() => Alert.alert.mock.calls[0][2][1].onPress());
    expect(Api.deleteContact).toBeCalled();
    expect(navigation.goBack).toBeCalled();
  });

  it('handle edit contact button correctly', async () => {
    Api.deleteContact.mockResolvedValue({
      data: {},
    });
    const route = {
      params: {
        id: 'b3abd640-c92b-11e8-b02f-cbfa15db428b',
      },
    };
    const component = (
      <Provider store={store}>
        <ContactDetailScreen navigation={navigation} route={route} />
      </Provider>
    );

    const {getByText} = render(component);

    const editContact = getByText('Ubah kontak');
    fireEvent.press(editContact.parent.parent);
    expect(navigation.navigate).toBeCalledWith(
      'CreateEditContact',
      expect.objectContaining({
        actionStatus: 'EDIT',
        contactDetail: expect.objectContaining({
          age: expect.any(Number),
          firstName: expect.any(String),
          id: expect.any(String),
          lastName: expect.any(String),
          photo: expect.any(String),
        }),
      }),
    );
  });
});
