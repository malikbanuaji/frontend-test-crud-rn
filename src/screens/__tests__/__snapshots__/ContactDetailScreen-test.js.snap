// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList loads contact with given id on route parameter (with picture) 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 72,
        "padding": 16,
      }
    }
    testID="header-bar"
  >
    <View
      style={
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
        testID="back-button"
      >
        <Icon
          name="chevron-left"
          size={24}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 16,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 8,
              "overflow": "hidden",
            },
          ]
        }
        testID="photo-big-button"
      >
        <Image
          height={225}
          source={
            Object {
              "uri": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
            }
          }
          style={
            Object {
              "height": 225,
              "width": 225,
            }
          }
          width={225}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Medium",
          "fontSize": 28,
          "marginHorizontal": 16,
        }
      }
    >
      Frodo Baggins
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Regular",
          "fontSize": 20,
          "marginHorizontal": 16,
        }
      }
    >
      24
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "borderColor": "#808080",
              "borderRadius": 4,
              "borderWidth": 1,
              "overflow": "hidden",
            },
            Object {
              "marginHorizontal": 16,
              "marginVertical": 8,
            },
          ],
          false,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Rubik-Medium",
                "fontSize": 16,
                "textAlign": "center",
              },
              Array [
                Object {
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 16,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
        >
          Ubah kontak
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#CA0B00",
              "borderRadius": 4,
              "overflow": "hidden",
            },
            Object {
              "marginHorizontal": 16,
              "marginVertical": 8,
            },
          ],
          false,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Rubik-Medium",
                "fontSize": 16,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 16,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
        >
          Hapus kontak
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ContactList loads contact with given id on route parameter 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 72,
        "padding": 16,
      }
    }
    testID="header-bar"
  >
    <View
      style={
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
        testID="back-button"
      >
        <Icon
          name="chevron-left"
          size={24}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 16,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 8,
              "overflow": "hidden",
            },
          ]
        }
        testID="photo-big-button"
      >
        <View
          style={
            Array [
              Object {
                "height": 225,
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#93b5e1",
                "borderRadius": 8,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Rubik-Medium",
                "fontSize": 48,
              }
            }
          >
            FB
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Medium",
          "fontSize": 28,
          "marginHorizontal": 16,
        }
      }
    >
      Frodo Baggins
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Regular",
          "fontSize": 20,
          "marginHorizontal": 16,
        }
      }
    >
      24
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "borderColor": "#808080",
              "borderRadius": 4,
              "borderWidth": 1,
              "overflow": "hidden",
            },
            Object {
              "marginHorizontal": 16,
              "marginVertical": 8,
            },
          ],
          false,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Rubik-Medium",
                "fontSize": 16,
                "textAlign": "center",
              },
              Array [
                Object {
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 16,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
        >
          Ubah kontak
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#CA0B00",
              "borderRadius": 4,
              "overflow": "hidden",
            },
            Object {
              "marginHorizontal": 16,
              "marginVertical": 8,
            },
          ],
          false,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Rubik-Medium",
                "fontSize": 16,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 16,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
        >
          Hapus kontak
        </Text>
      </View>
    </View>
  </View>
</View>
`;
